R/plumber.R

```diff
@@ -1020,7 +1028,20 @@ Plumber <- R6Class(
         if (is.null(node)){
           node <- list()
         }
-        node[[children[1]]] <- addPath(node[[children[1]]], children[-1], endpoint)
+        existing_endpoints <-
+          vapply(
+            node[which(names(node) == children[1])],
+            inherits,
+            logical(1),
+            "PlumberEndpoint")
+        if (any(existing_endpoints) && length(children) > 1) {
+          node <- c(
+            node[which(names(node) == children[1])][which(existing_endpoints)],
+            addPath(node[which(names(node) == children[1])][which(!existing_endpoints)], children, endpoint)
+          )
+        } else {
+          node[[children[1]]] <- addPath(node[[children[1]]], children[-1], endpoint)
+        }
```

Comment:
These are pretty complex statements. 

Do you mind adding a few comments / reasons for the code paths?

Thank you!

Reply:
Sure